pkg/client/k8s.go

```
@@ -298,6 +300,37 @@ func (k K8sClient) GetPVs(volNames []string, labelselector string) (*corev1.Pers
 	}, nil
 }
 
+// GetPvByCasType returns a list of PersistentVolumes based on cas-type slice
+// casTypes slice if is nil or empty, it returns all the PVs in the cluster.
+// casTypes slice if is not nil or not empty, it return the PVs with cas-types present in the slice.
+// labelselector takes the label(key+value) and makes an api call with this filter applied. Can be empty string if label filtering is not needed.
+func (k K8sClient) GetPvByCasType(casTypes []string, labelselector string) (*corev1.PersistentVolumeList, error) {
+	pvs, err := k.K8sCS.CoreV1().PersistentVolumes().List(context.TODO(), metav1.ListOptions{LabelSelector: labelselector})
+	if err != nil {
+		return nil, err
+	}
+
+	if len(casTypes) == 0 {
+		return pvs, nil
+	}
+
+	var list []corev1.PersistentVolume
+
+	for _, vol := range pvs.Items {
+		for _, casType := range casTypes {
+			if CSIProvisioner, ok := util.CasTypeToProvisionerMap[casType]; ok {
+				if vol.Spec.CSI != nil && vol.Spec.CSI.Driver == CSIProvisioner {
+					list = append(list, vol)
+				}
+			}
+		}
+	}
+
+	return &corev1.PersistentVolumeList{
```

(nitpick) Lets return an error if the `len(list)` is zero with something on the lines of `fmt.Errorf("couldn't find volumes of %s cas-type", casTypes)`.

This way the callers can just depend on the `error` values to detect something off and won't need to depend on the list object for more details.